### PR TITLE
Anonymous changed to Default role in upstream

### LINKED ISSUE #3750: Regression on User tests
The regression is due to role name change: https://github.com/theforeman/foreman/pull/3239

### PR DESCRIPTION
**Pull request by renzon**
close #3750

Problem solved with conditional until name change
affects all sat versions current under test

### CODE REVIEW

**Comment by renzon** [comment]
Downstream:
```console
==== 12 tests deselected by '-k UserRoleTestCase and test_positive_update' =====
=================== 1 passed, 12 deselected in 10.20 seconds ===================
```
Upstream:
```console
==== 12 tests deselected by '-k UserRoleTestCase and test_positive_update' =====
=================== 1 passed, 12 deselected in 7.59 seconds ====================
```

**Comment by oshtaier** [comment]
Master branch is always refer to the downstream, so it doesn't seem as good idea to add fixes specific for upstream here. Upstream can require a lot of different fixes due changes to functionality, so if we start to add all these workarounds we will quickly make our code unreadable and without possibility tracking 'dead' code.

I suggest to create separate branch for upstream that will contain all necessary fixes that we can merge with master after downstream require them. Of course, that will require pointing corresponding Jenkins jobs to use that branch. That approach has its own disadvantages as we will miss all new coverage in that branch or people can merge new things to it from time to time

**Comment by oshtaier** [comment]
That is only a suggestion and any other ideas are welcomed

File: tests/foreman/api/test_user.py [hunk 1]
```diff
@@ -193,7 +193,14 @@ def setUpClass(cls):
         """Create two roles and fetch the 'Anonymous' role."""
         super(UserRoleTestCase, cls).setUpClass()
         cls.roles = [entities.Role().create() for _ in range(2)]
-        roles = entities.Role().search(query={'search': 'name="Anonymous"'})
+        roles = entities.Role().search(query={'search': 'name="Default role"'})
+        if not roles:
+            # This is here because upstream changed name from "Anonumous" to
```
Comment (pgagne) [inline tests/foreman/api/test_user.py]: @renzon Anonumous -> Anonymous

**Comment by renzon** [comment]
@oshtaier I think it makes sense. This is currently been done for different sat versions. But I am including @sghai and @kbidarkar to check their opinion. If we go if branch will I probably need to inform @kbidarkar  to set up a new pipeline for upstream, right? Would this be the best approach?

File: tests/foreman/api/test_user.py [hunk 2]
```diff
@@ -193,7 +193,14 @@ def setUpClass(cls):
...
+            # "Default role" : Default role
+            # This can be removed once it get on all downstream versions
```
Comment (pgagne) [inline tests/foreman/api/test_user.py]: @renzon perhaps its better to say here "This can be removed once this is in all downstream versions"

**Comment by rochacbruno** [comment]
I think we should have a specific session in `robottelo.properties` file to store those variables, and in Jenkins job we need a text field when we can override that session.

I mean:

```
[tests]
user_default_role=Anonymous
```

Then in the code

```python
Role.search(query={"search": "name=%s" % settings.tests.user_default_role})
```

And when running in Jenkins we can add variables to be overwritten in that file.

**Comment by oshtaier** [comment]
@rochacbruno That is good enough option for that specific situation, but anyway we need to think about proper approach if we like to test upstream when we need to change few hundreds of lines for example

**Comment by sghai** [comment]
We have discussed about this in last automation meeting and decided to run downstream pipeline with 6.2.z. The only change in 6.2.z is pending on updating constant.py with sat version to support tools repo selection based on that.

Once this change is merged in 6.2.z, we are ready to make changes in master branch even on upstream issues. That's how we used to work earlier. So in short,  `master branch == upstream branch == 6.3`

@omaciel : please correct, If I'm missing something ?

**Comment by rochacbruno** [comment]
@oshtaier @sghai if `master branch == upstream branch == 6.3` so we need some way to have build based variables as I proposed in `robottelo.properties`

**Comment by oshtaier** [comment]
@rochacbruno @sghai Right, that is our usual way. Then we should not merge that PR till switch to 6.3 and then we do not need that `if` condition at all, just change the text. And it doesn't seem that we need to move something to properties, only in case we want to have more flexible way to change user role name in future

**Comment by sghai** [comment]
@oshtaier @rochacbruno, @renzon  : Yes, agreed. We should hold on this PR until 6.2.z will properly be in place.

**Comment by renzon** [comment]
Ok guys, I am going to edit this PR and change its text to match our decision

**Comment by sghai** [comment]
ACK..
I think we can go ahead and merge this PR now.

**Comment by oshtaier** [comment]
ACK

**Comment by renzon** [comment]
@oshtaier : could you merge it?

**Comment by oshtaier** [comment]
sure, just please change PR title

**Comment by renzon** [comment]
@oshtaier done

**Pull request merged**